.github/workflows/release.yml

```diff
@@ -0,0 +1,63 @@
+---
+name: Release and Publish
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup python 🐍
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Install dependencies 👽
+        run: python -m pip install ansible-core yq rst-to-myst
+
+      - name: Verify release 🗒️
+        env:
+          TAG_NAME: ${{ github.ref_name }}
+        run: |
+          echo "::warning::TODO: confirm tests pass for this commit"
+          tag_version="${TAG_NAME##*v}"
+          version="$(yq -r .version galaxy.yml)"
+          if [[ ${tag_version} != ${version} ]]; then
+            echo "::error file=galaxy.yml::🔴 Tag version '${tag_version}' does not match galaxy.yml version '${version}'"
+            exit 1
+          fi
+          echo "::notice::🟢 Version number verified: '$version'"
+
+      - name: Build 🛠️
+        run: ansible-galaxy collection build -v --force .
+
+      - name: Prepare release notes 📓
+        env:
+          CHANGELOG_URL: "${{ format('{0}/{1}/blob/{2}/CHANGELOG.rst', github.server_url, github.repository, github.ref_name) }}"
+          REF_NAME: "${{ github.ref_name }}"
+        run: |
+          rst2myst stream --no-sphinx CHANGELOG.rst | csplit - "/^## ${REF_NAME}/" '/^## v.*/'
+          mv xx01 release_notes.md
+          echo "See also: [CHANGELOG](${CHANGELOG_URL}#${REF_NAME//./-})" >>release_notes.md
+
+      - name: Release 🛸
+        uses: softprops/action-gh-release@v1
+        with:
+          files: |
+            *.tar.gz
+          body_path: release_notes.md
+          fail_on_unmatched_files: true
+          generate_release_notes: true
+
+  publish:
+    needs: release
```

Comment:
Should we make this a manual/workflow-dispatch step?

Reply:
I think that is a good idea for the first iteration. It would be great to have this be automated at some point, but we need to sort out our security posture first.

Reply:
Merging. Thank you for helping with this